artifactory_cleanup/cli.py

```diff
@@ -130,11 +147,27 @@ def main(self):
                 size(summary.artifacts_size),
             ]
             table.add_row(row)
+            output_json["policies"].append(
```

Comment:
I understand your request.
I had not made this choice because I think it is less optimized.

With the new solution, we change the execution time from O(n) to 3*O(n), since, with the Json output, we make 3 times the loop for the display. And we won't gain any memory since we will have to create the intermediate objects.

We gain in readability and factorization on the result construction but we lose in display optimization.

Moreover this loop already represents the display algo. The complexity is done via the construction of CleanupSummary

If you still prefer the new algo after this comment, I will do it.

Reply:
Thank you for the detailed explanation! I'll try to cover all your points

>We gain in readability and factorization

We do not, it less readable and maintainable solution because we have to keep two objects in our mind that we must maintain, but both with almost the same information, just a bit different purpose.

> I think it is less optimized.

One HTTP GET request levels out all our optimization like 100500 times :) Don't worry about performance here.

>O(n) to 3*O(n)

It's not quite right statement. Two cycles with `O(n)` complexity still has `O(n)`. Execution time can be increase twice, but not the complexity. 
We don't count how much time we go over the N elements, we just worried that we don't introduce inner loops or recursion that can increase **upper bound** for the complexity.


Let's change it to the suggested version with one element for holding "results" and two separate functions to ouptput it - one for printing, second for the file